src/remote.c

```diff
@@ -31,8 +31,11 @@ static int add_refspec_to(git_vector *vector, const char *string, bool is_fetch)
 	GITERR_CHECK_ALLOC(spec);
 
 	if (git_refspec__parse(spec, string, is_fetch) < 0) {
+                giterr_set(
+                        GITERR_INVALID,
+                        "'%s' is not a valid refspec.", string ? string : "(null)");
 		git__free(spec);
-		return -1;
+		return GIT_EINVALIDSPEC;
```

Comment:
That sounds reasonable to me. I misread refspec.c and was thinking that `git_refspec__parse` was just doing validation and was going to return a boolean; since it's actually populating a `git_refspec` instance it makes more sense to move the message setting there.

Should `git_refspec__parse` continue to return -1 on error, or should it return `GIT_EINVALIDSPEC`?

In addition the only tests touching this function are in `tests/network/refspecs.c`; should I add a test there to check that the error message is being set?